lib/languages/cpp.js

```diff
@@ -36,11 +36,19 @@ CppParser.prototype.parse_function = function(line) {
     );
 
     var matches = xregexp.exec(line, regex);
+    var retval;
     if(matches === null) {
+      var reg = xregexp('(?P<name>' + this.settings.fnIdentifier + ')' + '\\s*\\(\\s*(?P<args>.*?)\\)');
+      matches = xregexp.exec(line, reg);
+      if (matches === null) {
         return null;
+      }
     }
 
-    return [matches.name, matches.args, matches.retval];
+    var args = matches.args || null
+    retval = matches.retval || null
```

Comment:
Please move the declaration of retval down to its first assignment. As far as I can see it's not used before.